src/goto-checker/symex_bmc_incremental_one_loop.cpp

```diff
@@ -131,3 +135,29 @@ bool symex_bmc_incremental_one_loopt::resume(
 
   return should_pause_symex;
 }
+void symex_bmc_incremental_one_loopt::log_unwinding(unsigned unwind)
+{
+  const std::string unwind_num = std::to_string(unwind);
+  switch(output_ui)
+  {
+  case ui_message_handlert::uit::PLAIN:
```

Comment:
I know there are a few other places in the codebase where this sort of switch happens, so I'm not going to block this PR on this issue :-) But it's really ugly in general... I'm wondering if we could come up with an API extension to message handlers. Perhaps some pseudo code like this:
```
/// Individual implementations inside message handlers
keyValueMessagePair(string[] words, string value) {
/// words == ["current","unwinding"]
/// output would be:
/// "Current unwinding: value" for PLAIN
/// xmlt xml("current-unwinding"); xml.data = unwind_num; for XML
/// json_objectt json; json["currentUnwinding"] = json_numbert(unwind_num); for JSON
}
```
Obviously suitable C++ plumbing needed - but hopefully that explains the idea

Reply:
Yes I started having a go at making just a general way of logging structured data, but didn't want to hold up this PR for t